src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -270,13 +270,11 @@ public static void parseField(AbstractFieldMapper.Builder builder, String name,
             } else if (propName.equals("include_in_all")) {
                 builder.includeInAll(nodeBooleanValue(propNode));
                 iterator.remove();
-            } else if (propName.equals("postings_format")) {
-                String postingFormatName = propNode.toString();
-                builder.postingsFormat(parserContext.postingFormatService().get(postingFormatName));
+            } else if (propName.equals("postings_format") && parserContext.indexVersionCreated().before(Version.V_2_0_0)) {
+                // ignore in 1.x reject in 2.x
```

Comment:
Or actually just "ignore for old indexes" would probably be sufficient, since the version is clear from the condition.